ecs/context.go

```diff
@@ -166,78 +279,73 @@ func (h contextCreateAWSHelper) chooseProfile(profiles []string) (string, error)
 		return "", err
 	}
 	profile := options[selected]
-	if options[selected] == "new profile" {
-		suggestion := ""
-		if !contains(profiles, "default") {
-			suggestion = "default"
-		}
-		name, err := h.user.Input("profile name", suggestion)
-		if err != nil {
-			return "", err
-		}
-		if name == "" {
-			return "", fmt.Errorf("profile name cannot be empty")
-		}
-		return name, h.createProfile(name)
-	}
 	return profile, nil
 }
 
-func (h contextCreateAWSHelper) chooseRegion(region string, profile string) (string, error) {
-	suggestion := region
-
+func getRegion(profile string) (string, bool, error) {
+	if profile == "" {
+		profile = "default"
+	}
 	// only load ~/.aws/config
 	awsConfig := defaults.SharedConfigFilename()
 	configIni, err := ini.Load(awsConfig)
-
 	if err != nil {
 		if !os.IsNotExist(err) {
-			return "", err
+			return "", false, err
 		}
 		configIni = ini.Empty()
 	}
+
+	var f func(string) (string, string)
+	f = func(p string) (string, string) {
+		r := ""
+		section, err := configIni.GetSection(p)
+		if err == nil {
+			reg, err := section.GetKey("region")
+			if err == nil {
+				r = reg.Value()
+			}
+		}
+		if r == "" {
+			switch p {
+			case "":
+				return "us-east-1", ""
+			case "default":
+				return f("")
+			}
+			return f("default")
+		}
+		return r, p
+	}
+
 	if profile != "default" {
 		profile = fmt.Sprintf("profile %s", profile)
 	}
-	section, err := configIni.GetSection(profile)
-	if err != nil {
-		if !strings.Contains(err.Error(), "does not exist") {
-			return "", err
-		}
-		section, err = configIni.NewSection(profile)
+	region, p := f(profile)
+	return region, p == profile, nil
+}
+
+func (h contextCreateAWSHelper) chooseRegion(region string, profile string) (string, error) {
+	suggestion := region
+	if suggestion == "" {
+		region, _, err := getRegion(profile)
 		if err != nil {
 			return "", err
 		}
-	}
-	reg, err := section.GetKey("region")
-	if err == nil {
-		suggestion = reg.Value()
+		suggestion = region
 	}
 	// promp user for region
-	region, err = h.user.Input("Region", suggestion)
+	region, err := h.user.Input("Region", suggestion)
```

Comment:
Since we start having many scenarios in context creation, we should have some unit tests. That will require mocking `user.Input("Region", suggestion)`, that should not be too hard since this is already all behind an interface in prompt.go